src/permissions/Caveat.ts

```diff
@@ -222,6 +235,18 @@ export type ExtractCaveatValue<
   CaveatType extends string,
 > = ExtractCaveat<CaveatSpecifications, CaveatType>['value'];
 
+/**
+ * Determines whether a caveat specification has a decorator.
+ *
+ * @param specification - The caveat specification.
+ * @returns True if the caveat specification has a decorator, otherwise false.
+ */
+export function hasDecorator(
```

Comment:
The purpose of this is really to check that the value is a restricted method caveat specification, and the presence / absence of the decorator is an implementation detail. We should consider renaming this function and updating the docstring.